javacutil/src/org/checkerframework/javacutil/trees/TreeBuilder.java

```diff
@@ -140,6 +142,27 @@ public MemberSelectTree buildIteratorMethodAccess(ExpressionTree iterableExpr) {
         return iteratorAccess;
     }
 
+    /** Find the supertype of iteratorType that conforms to java.util.Iterator. */
+    private DeclaredType findIteratorSupertype(DeclaredType iteratorType) {
```

Comment:
Could you use `com.sun.tools.javac.code.Types#asSuper` instead of this method?  Or is there some reason not to use `asSuper`?

Reply:
Thanks! I had only looked in the model version of Types, not in the javac version. I changed this and at least the nullness tests continue to work.